simulation/core/node_tracker.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Class for tracking node information during simulation."""
+
+from typing import Optional, Tuple
+
+import numpy as np
+from core.last_used_ordered_set import LastUsedOrderedSet
+from core.utils import remove_padded_samples
+from numpy.typing import NDArray
+from sortedcollections import OrderedSet
+
+from streaming.base.spanner import Spanner
+
+
+class NodeTracker():
+    """Tracker for node information during simulation.
+
+    Args:
+        workers (int): The number of workers.
```

Comment:
Are these CPU workers? The `num_workers` parameter in Dataloader?

Reply:
yeah it's the same.